src/lib.rs

```diff
@@ -16,11 +25,77 @@
 use std::convert::From;
 use std::io::{Result, Write};
 use std::mem;
+use std::ops;
+use std::fmt;
```

Comment:
It's quite common to `use` incomplete paths in order to specify names in a more expressive and explicit manner. For example, `fmt::Result` can't even be imported into this namespace without shadowing the prelude `Result`. I see a few possibilities here:

1. Just order the current `use`-statements alphabetically.
2. Import all names directly that don't clash or are confusing without the module name (everything except `Result`, I guess).
3. Import all names and rename the clashing names with `use ... as ...;`.
4. What your comment currently says.